lib/mkdir-daily-rotate-file.js

```diff
@@ -12,6 +14,22 @@ util.inherits(MkdirDailyRotateFile, DailyRotateFile);
 MkdirDailyRotateFile.prototype.name = 'MkdirDailyRotateFile';
 
 MkdirDailyRotateFile.prototype.open = function open(cb) {
+    var now = new Date();
+    if (!this._stream ||
+      (this.maxsize && this._size >= this.maxsize) ||
+      (this._year < now.getFullYear() || this._month < now.getMonth() || this._date < now.getDate() || this._hour < now.getHours() || this._minute < now.getMinutes())
```

Comment:
That's what the upstream logic does.

This is far better then doing it for every single log() callsite.